azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -180,6 +185,70 @@ def disable(self):
+                for line in units.split('\n'):
+                    match = re.match(r'\s?(azure[^\s]*)\s?', line, re.IGNORECASE)
+                    if match is not None:
+                        azure_unit_name = match.group(1)
+                        azure_unit_description = line
+            except shellutil.CommandError as command_error:
+                message = "Failed to list systemd units: {0}".format(ustr(command_error))
+                logger.info(message)
+                add_event(op=WALAEventOperation.CGroupsInitialize, is_success=False, message=message, log_event=False)
+
+            if azure_unit_name is None:
+                return
+
+            azure_unit_location = "Unknown"
+            try:
+                azure_unit_location = SystemdCgroupsApi.get_unit_property(azure_unit_name, "Slice")
+            except Exception as exception:
+                message = "Failed to query Slice for {0}: {1}".format(azure_unit_name, ustr(exception))
+                logger.info(message)
+                add_event(op=WALAEventOperation.CGroupsInitialize, is_success=False, message=message, log_event=False)
+
+            message = "Found an Azure unit under {0}: {1}".format(azure_unit_location, azure_unit_description)
+            logger.info(message)
+            add_event(op=WALAEventOperation.CGroupsInitialize, message=message)
+
+        @staticmethod
+        def __collect_agent_unit_files_telemetry():
+            agent_unit_files = []
+            agent_service_name = get_osutil().get_service_name()
+            try:
+                fragment_path = SystemdCgroupsApi.get_unit_property(agent_service_name, "FragmentPath")
+                if fragment_path != "/lib/systemd/system/{0}.service".format(agent_service_name):
+                    agent_unit_files.append(fragment_path)
+            except Exception as exception:
+                message = "Failed to query the agent's FragmentPath: {0}".format(ustr(exception))
+                logger.info(message)
+                add_event(op=WALAEventOperation.CGroupsInitialize, is_success=False, message=message, log_event=False)
+
+            try:
+                drop_in_paths = SystemdCgroupsApi.get_unit_property(agent_service_name, "DropInPaths")
+                for path in drop_in_paths.split():
+                    agent_unit_files.append(path)
+            except Exception as exception:
+                message = "Failed to query the agent's DropInPaths: {0}".format(ustr(exception))
+                logger.info(message)
+                add_event(op=WALAEventOperation.CGroupsInitialize, is_success=False, message=message, log_event=False)
+
+            for unit_file in agent_unit_files:
+                try:
+                    with open(unit_file, "r") as file_object:
+                        message = "Found a custom unit file for the agent: {0}\n{1}".format(unit_file, file_object.read())
```

Comment:
This is very cool and much needed! This would give us a very good idea on how many VMs/Customers are modifying us without us knowing!

Reply:
yes, it should be helpful. sample logging:

```
2021-01-27T00:51:34.078443Z INFO ExtHandler ExtHandler Found an Azure unit under -.slice: azure.slice loaded active active Slice for Azure VM Agent and Extensions
2021-01-27T00:51:34.091817Z INFO ExtHandler ExtHandler Found a custom unit file for the agent: /etc/systemd/system/walinuxagent.service
#
# NOTE:
#   This file hosted on WALinuxAgent repository only for reference purposes.
#   Please refer to a recent image to find out the up-to-date systemd unit file.
#

[Unit]
Description=Azure Linux Agent

After=network-online.target cloud-init.service
Wants=network-online.target sshd.service sshd-keygen.service

ConditionFileIsExecutable=/usr/sbin/waagent
ConditionPathExists=/etc/waagent.conf

[Service]
Type=simple
ExecStart=/usr/bin/python3 -u /usr/sbin/waagent -daemon
Restart=always

[Install]
WantedBy=multi-user.target


2021-01-27T00:51:34.092579Z INFO ExtHandler ExtHandler Found a custom unit file for the agent: /etc/systemd/system/walinuxagent.service.d/10-azure-walinuxagent.system.conf
[Service]
Slice=azure.slice
CPUAccounting=yes

2021-01-27T00:51:34.093055Z INFO ExtHandler ExtHandler Found a custom unit file for the agent: /etc/systemd/system/walinuxagent.service.d/20-azure-walinuxagent.system.conf
[Service]
Slice=azure.slice
MemoryAccounting=yes

```